lib/line-numbers.js

```diff
@@ -0,0 +1,147 @@
+'use strict';
+
+const callsites = require('callsites');
+const sourceMapSupport = require('source-map-support');
+const parseTestSourceInFile = require('./test-ast');
+
+module.exports = {
+  parseLineNumbers,
+  parseLineNumbersInPath,
+  parseLineNumbersInPaths,
+  hasLineNumbersSuffix,
+  stripLineNumbersSuffix,
+  resolveEndLineNumberForTestInFile,
+  getLineNumberRangeForTestInFile,
+  isTestSelectedByLineNumbers
+};
+
+const NUMBER_REGEX = /^-?\d+$/;
+const RANGE_REGEX = /^(?<start>-?\d+)-(?<end>-?\d+)$/;
+const LINE_NUMBERS_REGEX = /^(-?\d+(-\d+)?,?)+$/
+const PATH_DELIMITER = ':';
+
+const parseNumber = string => parseInt(string, 10);
+const range = (start, end) => Array(end - start + 1).fill(start).map((e, i) => e + i);
+const distinctArray = array => [...new Set(array)];
+const removeAllWhitespace = string => string.replace(/\s/g, "");
+const pluralizeIf = (string, condition) => condition ? `${string}s` : string;
+const getLineNumbersSuffix = path => path.split(PATH_DELIMITER).pop();
+const sortNumbersAscending = array => {
+  const sorted = [...array];
+  sorted.sort((a, b) => a - b);
+  return sorted;
+};
+
+function parseLineNumbers(string) {
+  if (!string) {
+    throw new FormatError(string);
+  }
+
+  const parts = string.split(',');
+  const lineNumbers = parts.map(parsePart).flat();
```

Comment:
Our minimal supported version will (soon) be Node.js 10 LTS, which I don't think has `flat()` available in it.